### PR TITLE
Allow passing style and class props intended for the top-level div of an extension slot

### PR DESCRIPTION
**Pull request by brandones**
This makes it so there's no weird intermediate div you can't style

### CODE REVIEW

File: packages/esm-extension-manager/src/extension-slot-react.component.tsx [hunk 1]
```diff
@@ -9,6 +9,8 @@ import {
 export interface ExtensionSlotReactProps {
   extensionSlotName: string;
   children?: ReactNode;
+  style?: {};
```
Comment (FlorianRappl) [inline packages/esm-extension-manager/src/extension-slot-react.component.tsx:12]: What's that? Is that React.CSSProperties?

File: packages/esm-extension-manager/src/extension-slot-react.component.tsx [hunk 2]
```diff
@@ -9,6 +9,8 @@ import {
...
+  [key: string]: any; // rest of params passed to div
```
Comment (FlorianRappl) [inline packages/esm-extension-manager/src/extension-slot-react.component.tsx:13]: I would prefer to collect them in a prop, e.g., `params`. Reason: Right now we cannot distinguish; what if `style` or even `extensionSlotName` is the desired prop name (for whatever reason)? Alternatively, I'd prefix the hard-wired props to make the distinction more obvious.

For typing reasons I would not have `[key]: any`, but rather a generic `<T = {}>` to be something like

```ts
interface ExtensionSlotBaseProps {
  extensionSlotName: string;
  children?: ReactNode;
  style?: {};
}

type ExtensionSlotReactProps <T = {}> = ExtensionSlotBaseProps & T;
```

File: packages/esm-extension-manager/src/extension-slot-react.component.tsx [hunk 3]
```diff
@@ -40,8 +44,12 @@ export const ExtensionSlotReact: React.FC<ExtensionSlotReactProps> = ({
     ).then((names) => setExtensionNames(names));
   }, [extensionSlotName, moduleName]);
 
+  const divStyle = getIsUIEditorEnabled()
+    ? Object.assign(style || {}, { backgroundColor: "cyan" })
```
Comment (FlorianRappl) [inline packages/esm-extension-manager/src/extension-slot-react.component.tsx:48]: Any reason for mutating `style`? I'd never mutate a prop.

Just `{...style, backgroundColor: "cyan" }` - no need for the check and for all the braces (plus no mutation).

Comment (brandones) [reply packages/esm-extension-manager/src/extension-slot-react.component.tsx:48]: Oh yeah, this is absolutely the better way. Thanks!

**Comment by brandones** [comment]
Very useful feedback, thanks @FlorianRappl . Ready for re-review.

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks!

**Pull request merged**